r/R/csv.R

```diff
@@ -415,6 +415,16 @@ CsvReadOptions$create <- function(use_threads = option_use_threads(),
   )
 }
 
+readr_to_csv_write_options <- function(include_header,
+                                       batch_size = 1024L) {
+  assert_that(is_integerish(batch_size, n = 1, finite = TRUE), batch_size > 0)
```

Comment:
It will not pass the assertion, more specifically the `finite = TRUE` part.
```
> is_integerish(Inf, n = 1)
[1] TRUE 
> is_integerish(Inf, n = 1, finite = TRUE)
[1] FALSE
```

Reply:
If we disabled that check / assertion in both `readr_to_csv_write_options` and `CsvWriteOptions$create()`, we would surface the following error message from C++:
```
Error: Invalid: WriteOptions: batch_size must be at least 1: -2147483648
/Users/dragos/Documents/arrow/cpp/src/arrow/csv/writer.cc:291  options.Validate()
/Users/dragos/Documents/arrow/cpp/src/arrow/csv/writer.cc:433  MakeCSVWriter(output, table.schema(), options)
In addition: Warning message:
In csv___WriteOptions__initialize(list(include_header = include_header,  :
  NAs introduced by coercion to integer range
 ```

Reply:
Should we aim for a more informative error message here? 
```
> write_csv_arrow(my_tibble, csv_file, batch_size = Inf)
Error: is_integerish(x = batch_size, n = 1, finite = TRUE) is not TRUE
```
might not be the easiest to digest by a user encountering `rlang::is_integerish()` for the first time. We also offer no direction for further action.